sphinx/source/docs/dev_guide/bokehjs.rst

```diff
@@ -121,98 +121,231 @@ Use ``node make help`` to list all available commands.
 You can use ``tsc`` directly for error checking (e.g. in an IDE). However, don't use
 it for code emit, because we rely on AST transforms to produce viable library code.
 
-Testing
-~~~~~~~
+.. _devguide_bokehjs_development_testing:
 
-BokehJS testing is performed with the ``node make test`` command. You can run individual
-test suites with ``node make test:suite_name``. Known tests suites are:
+Test suites
+~~~~~~~~~~~
 
-* ``node make test:codebase``
-* ``node make test:defaults``
-* ``node make test:unit``
-* ``node make test:integration``
+BokehJS comes with its own suites of tests and testing framework. All tests for BokehJS
+use ``describe()`` and ``it()`` functions. Unit tests are written using Chai "expect"
+style. [TBD: How are style/setup different for visual integration tests?]
 
-The last two can be run with ``node make test:lib``. Unit and integration tests are
-run in a web browser (see requirements), which is started automatically with the
-right settings to guarantee consistent test results.
+To launch BokehJS tests, run ``node make test`` from within the
+:bokeh-tree:`bokehjs/test` directory.
 
-To review the visual tests' output, start BokehJS's devtools server:
+Instead of running all available BokehJS tests, you can also run individual test
+suites with ``node make test:suite_name``. Available tests suites are:
+
+* ``node make test:codebase``: Codebase tests checking file size limits
+* ``node make test:defaults``: Tests checking whether the defaults in Bokeh's
+  Python models match those of Bokeh's TypeScript models
+* ``node make test:unit``: Unit tests for BokehJS
+* ``node make test:integration``:
+  :ref:`Visual integration tests <devguide_bokehjs_development_visual_testing>`
+  comparing locally generated plots against a set of baseline files
+
+You can combine the last two test suites by running ``node make test:lib``.
+
+Unit and integration tests are run in a web browser (see requirements). The test suite
+automatically starts the web browser with the right settings to ensure consistent test
+results.
+
+.. _devguide_bokehjs_development_devtoolsserver:
+
+Testing with Devtool Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to tests run from the command line, you can also use the BokehJS's devtools
+server. Use this system to run tests and review the visual tests' output.
+
+First, start the devtools server with the following command:
 
 .. code-block:: sh
 
     $ node test/devtools server
     listening on 127.0.0.1:5777
 
-and navigate to ``/integration/report``. Devtools server can also be used to
-manually inspect and debug tests. For that, the following endpoints are available:
+Next, open the displayed server URL in a webbrowser and navigate to
```

Comment:
`web browser` is used instead of `webbrowser` elsewhere in the file.  I'd prefer one or the other rather than both.

Reply:
The first time I used the devtools server I used `127.0.0.1:5777` in my browser, received an error and immediately gave up, assuming it was broken.  Only later when I tried `127.0.0.1:5777/integration/report` did I realise it had been working fine all along.  So maybe we need to be a bit clearer not to use the server URL on its own to help those users who are as stupid as I am.